contracts/governance/Staking/Staking.sol

```diff
@@ -20,34 +20,38 @@ contract Staking is WeightedStaking{
         require(until > block.timestamp, "Staking::timestampToLockDate: staking period too short");
     
         //stake for the msg.sender if not specified otherwise
-        if(stakeFor == address(0))
+        if(stakeFor == address(0)) {
             stakeFor = msg.sender;
-        require(currentBalance(stakeFor, until) == 0, "Staking:stake: use 'increaseStake' to increase an existing staked position");
-        
+        }
+        //delegate for stakeFor if not specified otherwise
+        if(delegatee == address(0)) {
+            delegatee = stakeFor;
+        }
         //do not stake longer than the max duration
-        if (until > block.timestamp + MAX_DURATION)
+        if (until > block.timestamp + MAX_DURATION) {
             until = block.timestamp + MAX_DURATION;
-            
-        //retrieve the SOV tokens
-        bool success = SOVToken.transferFrom(msg.sender, address(this), amount);
-        require(success);
-        
-        //lock the tokens and update the balance by updating the user checkpoint
-        _increaseUserStake(stakeFor, until, amount);
-        
-        //increase staked token count until the new locking date
-        _increaseDailyStake(until, amount);
-        
-        //delegate to self in case no address provided
-        if(delegatee == address(0))
-            _delegate(stakeFor, stakeFor, until);
-        else
+        }
+
+        uint96 previousBalance = currentBalance(stakeFor, until);
+        //increase stake
+        _increaseStake(amount, stakeFor, until);
+
+        if (previousBalance == 0) {
+            //regular delegation if it's a first stake
             _delegate(stakeFor, delegatee, until);
-        
-        emit TokensStaked(stakeFor, amount, until, amount);
+        } else {
```

Comment:
nice, you're even allowing to change the delegate in the same transaction.

Reply:
thanks